app/actions/remote/post.ts

```diff
@@ -182,6 +188,59 @@ export async function createPost(serverUrl: string, post: Partial<Post>, files:
     return {data: true};
 }
 
+export const retryFailedPost = async (serverUrl: string, post: PostModel) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const timestamp = Date.now();
+        const apiPost = await post.toApi();
+        const newPost = {
+            ...apiPost,
+            props: {
+                ...apiPost.props,
+                failed: false,
+            },
+            id: '',
+            create_at: timestamp,
+            update_at: timestamp,
+            delete_at: 0,
+        } as Post;
+        const created = await client.createPost(newPost);
```

Comment:
From the look of this, the post will look as failed while the post gets posted. If the createPost takes long, or it fails, nothing changes in the front-end.

Am I missing something? Shouldn't we store the newPost without the failed state (and the new timestamp) and then do the client action (and handle the post one more time with the failed state or not after the client action is performed)?

Reply:
hm, I see what you mean.. what we could do is we could update the PostModel to remove the `failed` prop and update the timestamps as well.
Execute the request and in case of failure add back the `failed` prop..

Another option would be:
Remove the post and call `createPost` again...

Which approach are you inclining for?

Reply:
I feel removing and creating will create more artifacts (observables launching when the post is not there, everything reordering to show it, and then the post comes up again...). By updating the model, when the observables trigger, they will trigger just on the post (the user only perceive "color changes").

Whether we can refactor createPost to handle both cases, I leave up to you.

Reply:
cool.. I'll address it, probably ready by tomorrow.